packages/dds/merge-tree/src/client.ts

```diff
@@ -67,7 +67,16 @@ function elapsedMicroseconds(trace: Trace) {
     return trace.trace().duration * 1000;
 }
 
-export class Client {
+/**
+ * Emitted before this client's merge-tree normalizes its segments on reconnect, potentially
+ * ordering them. Useful for DDS-like consumers built atop the merge-tree to compute any information
+ * they need for rebasing their ops on reconnection.
+ *
+ * @internal
+ */
+export type IClientEvents = (event: "normalize", listener: () => void) => void;
```

Comment:
```suggestion
export type IClientEvents = (event: "normalize", listener: (target: IEventThisPlaceHolder) => void) => void;
```

i like to give the object to the event handler. this makes event handler re-usable without the need to create new function objects for each client that uses it